### ISSUE TITLE
Sort results

### ISSUE THREAD

**Issue by dwhieb** [dwhieb/Nisinoon#130]
- [x] https://adrianroselli.com/2021/04/sortable-table-columns.html
- [x] https://inclusive-components.design/data-tables/ (See section on sortable tables)
- [x] https://www.moesif.com/blog/technical/api-design/REST-API-Design-Filtering-Sorting-and-Pagination/
- [x] https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design

# Backend

- [x] Add a `sort` parameter whose value is a comma-separated list of field names preceded by a `+/-` to indicate ascending/descending.

# Frontend

- [x] `aria-sort`
  - [x] Place on the `<th>`.
  - [x] Values: `ascending` | `descending` | `none` (default) | `other`
  - [x] Don't use `none`; just remove the attribute.
  - [x] ~~Only one column should have the sort at a time.~~
- [x] Add sort icons (unsorted, ascending, descending).
  - [x] These should be contained within buttons, and the actual icon is `aria-hidden`.
  - [x] Add `aria-describedby`, pointing to the (hidden) `<caption>`.
  - [x] Change color on hover.
  - [x] Support dark mode.
- [x] Add JS functionality to adjust querystring and submit search to server.
  - [x] Third click on column removes it from the sort directives.
  - [x] The sort priority needs to be retained during parsing/serialization of the sort directives.
- [x] Indicate sort priority.
  - [x] `data-priority` on `<th>` to indicate sort priority.
  - [x] Hidden span within the `<th>` showing the sort priority.
  - [x] Sort priority badges on/near/above `<th>`.
